### DISCUSSION TITLE
Deletion for all sites exposed to users without permission

### DISCUSSION THREAD

**Discussion by zsavajji** [craftcms/cms#9956] [Ideas] [answered]
Hi,
i'm building a multi-country / multi-language Craft website, and i found out a bug happening in the deletion of entries.

I have an "admin" role which can add / edit / delete entries for all countries and languages, and some "regional" editors which can only edit a subset of this localized entries. 
After some testing i found that the localized editor user can also delete entries of other sites, excluded from its permissions.

The delete button of the editor account should only be a "Delete for site" action, but the account has also access to the regular delete button. 

<img width="1041" alt="FR_permissions" src="https://user-images.githubusercontent.com/6460348/136751662-687a9e44-08e9-405a-ad3f-890d6fee308b.png">

<img width="641" alt="FR_delete bug" src="https://user-images.githubusercontent.com/6460348/136751614-81271c37-7be1-4a50-9558-a3e0114657ec.png">

I'm on latest Craft PRO 3.7.16, MySQL / php8

Thanks!

**Comment by brandonkelly** [comment, marked as answer]
The permission is for the main “Delete” action, which is a global operation, so this is working as expected.

We could consider adding a permission specifically for the “Delete for site” action, for sites where the Propagation Method is set to “Let each entry choose which sites it should be saved to”.

**Reply by zsavajji** [reply]
No, it's not, I think i explained myself badly. 
If i give a user the permission to edit some sites only, this user should only see the "delete for site" button, since it should not apply global operations.
My restricted user can, with this set of permissions, also delete entries for the US/IT websites, using the global "Delete" button, which it should not have access to.

![image](https://user-images.githubusercontent.com/6460348/136974307-9f8ebd53-773c-4fab-b766-17ed801d4edc.png)

Let me know if i missed something :)

**Reply by brandonkelly** [reply]
I understand you don’t want users without access to all sites to be able to delete entries globally. I’m saying that you are misunderstanding the permission though. “Delete entries” really does mean “Delete entries globally”. So the permission is working as intended. Your issue would be solved if we were to introduce a _new_ (lesser) permission for only deleting entries for a specific site.

**Reply by zsavajji** [reply]
Oh, ok! Now i get it 😄!
Yes, that would solve the issue, and would also be reasonable for a true multi site ownership, since every single site editor would have complete control over its own entries.

Thank you
